openmctStaticServer/dictionary-plugin.js

```diff
@@ -1,79 +1,127 @@
+                telemetryEntry.values.forEach(function (value) {
                     if (value.units !== undefined) {
                         value.name += " (" + value.units + ")";
                     }
                 });
                 return {
                     identifier: identifier,
-                    name: measurement.name,
-                    type: 'icubsensor.telemetry',
+                    name: telemetryEntry.name,
+                    type: this.telemetryEntryType,
                     telemetry: {
-                        values: measurement.values
+                        values: telemetryEntry.values
                     },
-                    location: DOMAIN_OBJECTS_NAMESPACE + ':' + ROOT_DOMAIN_OBJECT_KEY
+                    location: identifier.namespace + ':' + dictionary.key
                 };
             }
-        });
+        }.bind(this));
     }
-};
+}
 
 var compositionProvider = {
     appliesTo: function (domainObject) {
-        return domainObject.identifier.namespace === DOMAIN_OBJECTS_NAMESPACE &&
-               domainObject.type === 'folder';
+        return (domainObject.identifier.namespace === ICUBTELEMETRY_DOMAIN_OBJECTS_NAMESPACE
+            || domainObject.identifier.namespace === WALKINGCTRLTELEMETRY_DOMAIN_OBJECTS_NAMESPACE)
+            && domainObject.type === 'folder';
     },
     load: function (domainObject) {
-        return getDictionary()
+        return getDictionary(domainObject.identifier)
             .then(function (dictionary) {
-                return dictionary.measurements.map(function (m) {
+                return dictionary.telemetryEntries.map(function (m) {
                     return {
-                        namespace: DOMAIN_OBJECTS_NAMESPACE,
+                        namespace: domainObject.identifier.namespace,
                         key: m.key
                     };
                 });
             });
     }
 };
 
-function DictionaryPlugin() {
+function DictionaryPlugin(telemServerHost,telemServerPort) {
     return function install(openmct) {
-        openmct.objects.addRoot({
-            namespace: DOMAIN_OBJECTS_NAMESPACE,
-            key: ROOT_DOMAIN_OBJECT_KEY
+        telemServerAddress.host = telemServerHost;
+        telemServerAddress.port = telemServerPort;
+
+        openmct.types.addType(ICUBTELEMETRY_DOMAIN_OBJECTS_TYPE, {
+            name: 'iCub Sensor Telemetry Entry',
+            description: 'Telemetry entry from one or multiple iCub sensors published on a single port.',
+            cssClass: 'icon-telemetry'
         });
 
-        openmct.objects.addProvider(DOMAIN_OBJECTS_NAMESPACE, objectProvider);
+        openmct.types.addType(WALKINGCTRLTELEMETRY_DOMAIN_OBJECTS_TYPE, {
+            name: 'Walking Controller Telemetry Entry',
+            description: 'Telemetry entry from the Walking Controller related data published on a single port (e.g. internal variables), structured as a vector collection map.',
```

Comment:
Again, not a blocking comment but for the future could it make sense to gradually try to move this information from the `.js` files (that most users will not know how to modify) to the `.json` files (that instead act as a "configuration" file)?

Reply:
Good point, and this is quite easy. Will do.

Reply:
Addressed in #96 .